crates/cast/tests/e2e/account/delete.rs

```diff
@@ -0,0 +1,70 @@
+use crate::helpers::constants::DEVNET_OZ_CLASS_HASH;
+use crate::helpers::fixtures::default_cli_args;
+use crate::helpers::runner::runner;
+use indoc::indoc;
+
+#[tokio::test]
+pub async fn test_account_network_should_exist() {
+    let mut args = default_cli_args();
+    args.append(&mut vec![
+        "account",
+        "delete",
+        "--name",
+        "user99",
+        "--network",
+        "goerli0-network",
+    ]);
+
+    let snapbox = runner(&args);
+
+    snapbox.assert().stderr_matches(indoc! {r#"
+    command: account delete
+    error: No accounts defined for network goerli0-network
+    "#});
+}
+
+#[tokio::test]
+pub async fn test_account_delete_should_exist() {
+    let mut args = default_cli_args();
+    args.append(&mut vec!["account", "delete", "--name", "user99"]);
+
+    let snapbox = runner(&args);
+
+    snapbox.assert().stderr_matches(indoc! {r#"
+    command: account delete
+    error: Account with name user99 does not exist
+    "#});
+}
+
+#[tokio::test]
+pub async fn test_account_delete_happy_case() {
+    // First, create a dummy account
+    let mut args = default_cli_args();
+    args.append(&mut vec![
+        "account",
+        "create",
+        "--name",
+        "user99",
+        "--salt",
+        "0x1",
+        "--class-hash",
+        DEVNET_OZ_CLASS_HASH,
+    ]);
```

Comment:
let's not use sncast account create for this - we do not need to create users on devnet chain to test this functionality. Let's add a helper that just writes some random data to the file instead, and use it here. Preferably, this would be done on some new, temporary accounts file (and not tests/data/accounts/accounts.json) to avoid messing up with the rest of the tests